src/dotnet-interactive.Tests/StdIoKernelConnectorTests.cs

```diff
@@ -135,6 +162,81 @@ public async Task it_throws_if_there_is_no_remote_subkernel_with_the_specified_n
             await action.Should().ThrowAsync<InvalidOperationException>();
         }
 
+        [Fact]
+        public async Task value_sharing_between_remote_kernels_works()
+        {
+            var connector = CreateConnector();
+            using var rootProxyKernel = await connector.CreateRootProxyKernelAsync();
+
+            var result = await rootProxyKernel.SendAsync(new RequestKernelInfo());
+            var kernelInfos = result.Events.OfType<KernelInfoProduced>().Select(e => e.KernelInfo);
+
+            using var _ = new AssertionScope();
+
+            var csharpKernelInfo = kernelInfos.Should().ContainSingle(i => i.LanguageName == "C#").Which;
```

Comment:
The intermediate sanity check assertions are probably not useful. They make for a lot more test code and make the intent of the test harder to grasp.

Assume happy path until you get to the assertions that prove the scenario named by the test.